lxd/db/cluster/profiles.mapper.go

```diff
@@ -121,42 +122,91 @@ func ProfileExists(ctx context.Context, tx *sql.Tx, project string, name string)
 
 // GetProfiles returns all available profiles.
 // generator: profile GetMany
-func GetProfiles(ctx context.Context, tx *sql.Tx, filter ProfileFilter) ([]Profile, error) {
+func GetProfiles(ctx context.Context, tx *sql.Tx, filters ...ProfileFilter) ([]Profile, error) {
 	var err error
 
 	// Result slice.
 	objects := make([]Profile, 0)
 
 	// Pick the prepared statement and arguments to use based on active criteria.
 	var sqlStmt *sql.Stmt
-	var args []any
+	args := make([]any, 0, DqliteMaxParams)
+	queryParts := [2]string{}
 
-	if filter.Project != nil && filter.Name != nil && filter.ID == nil {
-		sqlStmt = Stmt(tx, profileObjectsByProjectAndName)
-		args = []any{
-			filter.Project,
-			filter.Name,
-		}
-	} else if filter.Project != nil && filter.ID == nil && filter.Name == nil {
-		sqlStmt = Stmt(tx, profileObjectsByProject)
-		args = []any{
-			filter.Project,
-		}
-	} else if filter.Name != nil && filter.ID == nil && filter.Project == nil {
-		sqlStmt = Stmt(tx, profileObjectsByName)
-		args = []any{
-			filter.Name,
-		}
-	} else if filter.ID != nil && filter.Project == nil && filter.Name == nil {
-		sqlStmt = Stmt(tx, profileObjectsByID)
-		args = []any{
-			filter.ID,
-		}
-	} else if filter.ID == nil && filter.Project == nil && filter.Name == nil {
+	if len(filters) == 0 {
 		sqlStmt = Stmt(tx, profileObjects)
-		args = []any{}
-	} else {
-		return nil, fmt.Errorf("No statement exists for the given Filter")
+	}
+
+	for i, filter := range filters {
+		if filter.Project != nil && filter.Name != nil && filter.ID == nil {
+			args = append(args, []any{filter.Project, filter.Name}...)
+			if len(filters) == 1 {
```

Comment:
This `len(filters)` is being called one for each filter, you could call it once and store the result in a var before the loop starts.